pkg/server/server.go

```diff
@@ -159,6 +160,23 @@ func (s *Server) gitopsProxyEnabled() bool {
 	return s.GitOpsProxyConfig != nil
 }
 
+func (s *Server) GetKubeVersion() string {
```

Comment:
I am ok with adding this function here, but we would need to check with @spadgett what he thinks. If that is ok, `actions.KubeVersionGetter` should be moved in this package, and `actoin.GetKubeVersion()` should be integrated in this function.

Reply:
@pedjak `action` package is being imported in `server` package already. If we move `KubeVersionGetter` to `server`, there would be cyclic dependency issue.

Reply:
how about this:

```diff
diff --git a/pkg/helm/chartproxy/proxy.go b/pkg/helm/chartproxy/proxy.go
index 8bf5b3440..dc024214d 100644
--- a/pkg/helm/chartproxy/proxy.go
+++ b/pkg/helm/chartproxy/proxy.go
@@ -1,6 +1,7 @@
 package chartproxy
 
 import (
+	"github.com/openshift/console/pkg/version"
 	"helm.sh/helm/v3/pkg/chartutil"
 	"helm.sh/helm/v3/pkg/repo"
 	"k8s.io/client-go/dynamic"
@@ -8,8 +9,6 @@ import (
 	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/rest"
 	"k8s.io/klog"
-
-	"github.com/openshift/console/pkg/helm/actions"
 )
 
 type proxy struct {
@@ -48,7 +47,7 @@ func coreClientProvider(p *proxy) error {
 
 var defaultOptions = []ProxyOption{dynamicKubeClientProvider, coreClientProvider}
 
-func New(k8sConfig RestConfigProvider, kubeVersionGetter actions.KubeVersionGetter, opts ...ProxyOption) (Proxy, error) {
+func New(k8sConfig RestConfigProvider, kubeVersionGetter version.KubeVersionGetter, opts ...ProxyOption) (Proxy, error) {
 	config, err := k8sConfig()
 
 	if err != nil {
diff --git a/pkg/helm/handlers/handlers.go b/pkg/helm/handlers/handlers.go
index 93dac8858..0b1f34e86 100644
--- a/pkg/helm/handlers/handlers.go
+++ b/pkg/helm/handlers/handlers.go
@@ -15,9 +15,10 @@ import (
 	"github.com/openshift/console/pkg/helm/actions"
 	"github.com/openshift/console/pkg/helm/chartproxy"
 	"github.com/openshift/console/pkg/serverutils"
+	"github.com/openshift/console/pkg/version"
 )
 
-func New(apiUrl string, transport http.RoundTripper, kubeversionGetter actions.KubeVersionGetter) *helmHandlers {
+func New(apiUrl string, transport http.RoundTripper, kubeversionGetter version.KubeVersionGetter) *helmHandlers {
 	h := &helmHandlers{
 		ApiServerHost:           apiUrl,
 		Transport:               transport,
diff --git a/pkg/server/kube_version.go b/pkg/server/kube_version.go
index 734910d02..475e616e4 100644
--- a/pkg/server/kube_version.go
+++ b/pkg/server/kube_version.go
@@ -1,16 +1,30 @@
-package actions
+package server
 
 import (
 	"errors"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
+	"k8s.io/klog"
 )
 
-type KubeVersionGetter interface {
-	GetKubeVersion() string
+func (s *Server) GetKubeVersion() string {
+	if s.KubeVersion != "" {
+		return s.KubeVersion
+	}
+	config := &rest.Config{
+		Host:      s.K8sProxyConfig.Endpoint.String(),
+		Transport: s.K8sClient.Transport,
+	}
+	kubeVersion, err := kubeVersion(config)
+	if err != nil {
+		kubeVersion = ""
+		klog.Warningf("Failed to get cluster k8s version from api server %s", err.Error())
+	}
+	s.KubeVersion = kubeVersion
+	return s.KubeVersion
 }
 
-func GetKubeVersion(config *rest.Config) (string, error) {
+func kubeVersion(config *rest.Config) (string, error) {
 	client, err := discovery.NewDiscoveryClientForConfig(config)
 	if err != nil {
 		return "", err
diff --git a/pkg/server/server.go b/pkg/server/server.go
index 0d1b3cb6a..c1d9201f4 100644
--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -9,20 +9,16 @@ import (
 	"html/template"
 	"io"
 	"io/ioutil"
-	"k8s.io/client-go/rest"
 	"net/http"
 	"net/url"
 	"os"
 	"path"
 	"strings"
 
-	"k8s.io/klog"
-
 	"github.com/coreos/dex/api"
 	"github.com/coreos/pkg/health"
 
 	"github.com/openshift/console/pkg/auth"
-	"github.com/openshift/console/pkg/helm/actions"
 	helmhandlerspkg "github.com/openshift/console/pkg/helm/handlers"
 	"github.com/openshift/console/pkg/proxy"
 	"github.com/openshift/console/pkg/serverutils"
@@ -160,23 +156,6 @@ func (s *Server) gitopsProxyEnabled() bool {
 	return s.GitOpsProxyConfig != nil
 }
 
-func (s *Server) GetKubeVersion() string {
-	if s.KubeVersion != "" {
-		return s.KubeVersion
-	}
-	config := &rest.Config{
-		Host:      s.K8sProxyConfig.Endpoint.String(),
-		Transport: s.K8sClient.Transport,
-	}
-	kubeVersion, err := actions.GetKubeVersion(config)
-	if err != nil {
-		kubeVersion = ""
-		klog.Warningf("Failed to get cluster k8s version from api server %s", err.Error())
-	}
-	s.KubeVersion = kubeVersion
-	return s.KubeVersion
-}
-
 func (s *Server) HTTPHandler() http.Handler {
 	mux := http.NewServeMux()
 
diff --git a/pkg/version/version.go b/pkg/version/version.go
index dc5dae858..fa1850572 100644
--- a/pkg/version/version.go
+++ b/pkg/version/version.go
@@ -3,3 +3,7 @@ package version
 // version.Version should be provided at build time with
 //-ldflags "-X github.com/openshift/console/version.Version $GIT_TAG"
 var Version string
+
+type KubeVersionGetter interface {
+	GetKubeVersion() string
+}

```

@spadgett would this be ok for you?

Reply:
@pedjak I'm fine with that change.

Reply:
Done